brian2/codegen/permutation_analysis.py

```diff
@@ -33,10 +34,11 @@ def check_for_order_independence(statements, variables, indices):
     # only from the values of the derived indices. In the most common case of Synapses, the main index would be
     # the synapse index, and the derived indices would be pre and postsynaptic indices (which can be repeated).
     main_index_variables = set([v for v in variables
-                                if (indices[v] in ('_idx', '0')
-                                    or getattr(variables[indices[v]],
-                                               'unique',
-                                               False))])
+                                if (not isinstance(variables[v], Constant) and
+                                    (indices[v] in ('_idx', '0')
+                                     or getattr(variables[indices[v]],
+                                                'unique',
+                                                False)))])
```

Comment:
I agree this is not really clear. This all works because the `different_index_variables` are considered as permutation-independent (which is correct for a constant) and a constant can never get into the permutation-dependent collection because it does never appear on the LHS. But it would be probably better to have constants in a third set that is neither `main_index_variables` nor `different_index_variables`. That said, looking at the code I think scalar variables are still handled incorrectly, e.g. a `pre` statement `v_post = scalar_syn` will unnecessarily trigger a warning. I'll try to fix (and test) that tomorrow.